bitswap/client/internal/sessionmanager/sessionmanager.go

```diff
@@ -157,6 +157,9 @@ func (sm *SessionManager) GetNextSessionID() uint64 {
 // their contents. If the caller needs to preserve a copy of the lists it
 // should make a copy before calling ReceiveFrom.
 func (sm *SessionManager) ReceiveFrom(ctx context.Context, p peer.ID, blks []cid.Cid, haves []cid.Cid, dontHaves []cid.Cid) {
+	// Send CANCEL to all peers with want-have / want-block
+	sm.peerManager.SendCancels(ctx, blks)
```

Comment:
Perhaps explain why this sits here now, as I can imagine a developer undoing this in a couple of years when trying to optimize.

Reply:
Done